target_chains/solana/programs/solana-receiver/src/lib.rs

```diff
@@ -1,15 +1,91 @@
-use anchor_lang::prelude::*;
+pub mod error;
+pub mod state;
 
-declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");
+use {
+    wormhole::Chain::{
+        self,
+        Solana,
+    },
+    state::AnchorVaa,
+    anchor_lang::prelude::*,
+    pyth_wormhole_attester_sdk::PriceAttestation,
+};
+
+use crate::error::ReceiverError::*;
+
+declare_id!("5dXnHcDXdXaiEp9QgknCDPsEhJStSqZqJ4ATirWfEqeY");
 
 #[program]
-pub mod solana_receiver {
+pub mod pyth_solana_receiver {
     use super::*;
 
-    pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
+    pub fn decode_posted_vaa(ctx: Context<DecodePostedVaa>) -> Result<()> {
+        let posted_vaa = &ctx.accounts.posted_vaa.payload;
+        let attestation = PriceAttestation::deserialize(posted_vaa.as_slice())
+            .map_err(|_| DeserializeVAAFailed)?;
+
+        msg!("product_id: {}", attestation.product_id);
+        msg!("price_id: {}", attestation.price_id);
+        msg!("price: {}", attestation.price);
+        msg!("conf: {}", attestation.conf);
+        msg!("ema_price: {}", attestation.ema_price);
+        msg!("ema_conf: {}", attestation.ema_conf);
+        msg!("num_publishers: {}", attestation.num_publishers);
+        msg!("publish_time: {}", attestation.publish_time);
+        msg!("attestation_time: {}", attestation.attestation_time);
+
         Ok(())
     }
 }
 
 #[derive(Accounts)]
-pub struct Initialize {}
+pub struct DecodePostedVaa<'info> {
+    #[account(mut)]
+    pub payer:          Signer<'info>,
+    #[account(constraint = Chain::from(posted_vaa.emitter_chain) == Solana @ EmitterChainNotSolana)]
```

Comment:
emitter_chain : Pythnet or Solana

Reply:
You also need to check the magic number in the vaa

Reply:
the emitter could actually be both Pythnet and Solana (we still have solana as backup which can be used for solana again :)))

Reply:
Right, so I added check for magic number and emitter_chain is one of Pythnet and Solana.